### PR TITLE
Util: Signature Code Cleanup (new)

### PR DESCRIPTION
**Pull request by holgerd77**
Replaces #1940 after #1943 merge

### CODE REVIEW

**Comment by holgerd77** [comment]
@jochem-brouwer I have some last test failues here in the VM `EIP-3074` Auth Call tests which I thought would be an easy fix but apparently it's not, tried in the last commit with no success yet.

Could you eventually have a look? Guess this is a lot easier for you to spot than for me.

**Comment by jochem-brouwer** [comment]
@holgerd77 Cool, yeah, will do!

**Comment by gabrocheleau** [comment]
I think I just found the issue, pushing a fix.

**Comment by gabrocheleau** [comment]
Alright, fixed the issue and simplified the logic in `signature` in commit https://github.com/ethereumjs/ethereumjs-monorepo/pull/1945/commits/ec0027482e559cae6defedae1bae78b66ae3e061

**Review by gabrocheleau — APPROVED** [review]

**Review by gabrocheleau — APPROVED** [review]

File: packages/util/src/signature.ts [hunk 1]
```diff
@@ -22,12 +22,11 @@ export function ecsign(msgHash: Buffer, privateKey: Buffer, chainId?: bigint): E
...
   const s = Buffer.from(signature.slice(32, 64))
 
-  if (chainId === undefined) {
-    const v = BigInt(recovery + 27)
-    return { r, s, v }
-  }
+  const v =
+    chainId === undefined
+      ? BigInt(recovery + 27)
+      : BigInt(recovery + 35) + BigInt(chainId) * BigInt(2)
```
Comment (jochem-brouwer) [inline packages/util/src/signature.ts:28]: I'm not sure why this fixes it. These formulas look the same?

Comment (holgerd77) [reply packages/util/src/signature.ts:28]: This actually doesn't fix anything, I just tested to checkout the commit before and tests are still passing. I guess Gabriel meant the changes in the commit before to fix the situation.

Will re-read to try to give myself a confirmation on that.

**Comment by holgerd77** [comment]
Thanks @gabrocheleau for fixing 👍 🙂 and the direct approval, will merge.

**Pull request merged**